airbyte-integrations/connectors/source-exchange-rates/source_exchange_rates/source.py

```diff
@@ -110,7 +110,17 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
             logger.info(f"Ping response code: {status}")
             if status == 200:
                 return True, None
-            return False, resp.text
+            # When API requests is sent but the requested data is not available or the API call fails
+            # for some reason, a JSON error is returned.
+            # https://exchangeratesapi.io/documentation/#errors
+            error = resp.json().get("error")
+            code = error.get("code")
+            message = error.get("message")
+            # If code is base_currency_access_restricted, error is caused by switching base currency while using free
+            # plan
+            if code == "base_currency_access_restricted":
```

Comment:
Why do we need to distinguish exactly `base_currency_access_restricted` code? Like there are other "text codes" and what about them?

Reply:
According to issue linked to this PR we should add more clearer message, when user try to switch base currency when he has free plan. API returns error with code base_currency_access_restricted in this case. And that why checking  exactly for this code was added.